drivers/nutdrv_qx_masterguard.c

```diff
@@ -0,0 +1,1049 @@
+	{ "ups.alarm",			0,	NULL,	"Q3\r",	"",	69,	'(',	"",	63,	63,	NULL,	QX_FLAG_QUICK_POLL,	NULL,	NULL,	masterguard_status },	/* T6: problems in parallel operation mode */
+	/* part of a parallel set */																			/* T5: is part of a parallel set */
+	{ "ups.status",			0,	NULL,	"Q3\r",	"",	71,	'(',	"",	65,	65,	NULL,	QX_FLAG_QUICK_POLL,	NULL,	NULL,	masterguard_status },	/* T4: Battery: end of service life */
+	{ "ups.alarm",			0,	NULL,	"Q3\r",	"",	71,	'(',	"",	66,	66,	NULL,	QX_FLAG_QUICK_POLL,	NULL,	NULL,	masterguard_status },	/* T3: battery connected */
+	{ "ups.test.result",		0,	NULL,	"Q3\r",	"",	71,	'(',	"",	67,	69,	NULL,	QX_FLAG_QUICK_POLL,	NULL,	NULL,	masterguard_status },	/* T210: Test Status */
+
+	/*
+	 * > [GS,XX\r]
+	 * < [(XX,ii p a\r]
+	 *    01234567890
+	 *    0         1
+	 *    (00,00 0 1
+	 */
+	/* type				flags		rw			command		answer	len	leading	value	from	to	dfl	qxflags					precmd				preans				preproc */
+	/* ups.id obtained via WH */
+	{ "ups.id",			0,		NULL,			"SS%02d--,XX\r","",	0,	'\0',	"",	0,	0,	"d",	QX_FLAG_SETVAR | QX_FLAG_RANGE,		masterguard_add_slaveaddr,	masterguard_new_slaveaddr,	masterguard_set_slaveaddr },
+	{ "battery.packs",		ST_FLAG_RW,	masterguard_r_batpacks,	"GS,XX\r",	"",	0,	'(',	"",	7,	7,	"%.0f",	QX_FLAG_SEMI_STATIC | QX_FLAG_RANGE,	masterguard_add_slaveaddr,	NULL,				NULL },
+	{ "battery.packs",		0,		NULL,			"SS--%1d-,XX\r","",	0,	'\0',	"",	0,	0,	"d",	QX_FLAG_SETVAR | QX_FLAG_RANGE,		masterguard_add_slaveaddr,	NULL,				masterguard_setvar },
+/*!! which QX_FLAGs to use?? (changed by instcmd) */
+	{ "ups.beeper.status",		0,		NULL,			"GS,XX\r",	"",	11,	'(',	"",	9,	9,	NULL,	QX_FLAG_SEMI_STATIC,			masterguard_add_slaveaddr,	NULL,				masterguard_beeper_status },
+	/* set with beeper.{en,dis}able */
+
+	/*
+	 * > [GBS,XX\r]
+	 * < [(XX,CCC hhhh HHHH AAAA BBBB DDDD EEE SS.SS\r]
+	 *    0123456789012345678901234567890123456789012
+	 *    0         1         2         3         4
+	 *    (00,100 0017 0000 0708 0712 0994 115 02.28
+	 *    (01,000 0000 0360 0708 0712 0994 076 02.30
+	 */
+	/* type			flags	rw	command		answer	len	leading	value	from	to	dfl	qxflags	precmd				preans	preproc */
+	{ "battery.charge",	0,	NULL,	"GBS,XX\r",	"",	43,	'(',	"",	4,	6,	"%.0f",	0,	masterguard_add_slaveaddr,	NULL,	NULL },
+	/* 
+	 * hhhh: hold time (minutes)
+	 * HHHH: recharge time to 90% (minutes)
+	 * AAAA: Ageing factor (promilles)
+	 * BBBB: Ageing factor time dependant (promilles)
+	 * DDDD: Ageing factor cyclic use (promilles)
+	 * EEE: Calibration factor (percent)
+	 * SS.SS: Actual battery cell voltage
+	*/
+
+	/*
+	 * > [GSN,XX\r]
+	 * < [(XX,SSN=SSSSSSSSSSSnnnnn\r]
+	 *    0123456789012345678901234
+	 *    0         1         2
+	 *    (00,SSN=6A1212      2782
+	 */
+	/* type			flags	rw	command		answer	len	leading	value	from	to	dfl	qxflags				precmd				preans	preproc */
+	{ "device.part",	0,	NULL,	"GSN,XX\r",	"",	25,	'(',	"",	8,	18,	"%s",	QX_FLAG_STATIC | QX_FLAG_TRIM,	masterguard_add_slaveaddr,	NULL,	NULL },
```

Comment:
@aquette @clepple : Any suggestions for standard field naming?

Reply:
`device.part` is listed in the 2.7.4 names table, or are you referring to something else?